src/System.Private.CoreLib/shared/System/String.Comparison.cs

```diff
@@ -566,7 +566,12 @@ public bool EndsWith(string value, StringComparison comparisonType)
                     return CompareInfo.Invariant.IsSuffix(this, value, GetCaseCompareOfComparisonCulture(comparisonType));
 
                 case StringComparison.Ordinal:
-                    return this.Length < value.Length ? false : (CompareOrdinalHelper(this, this.Length - value.Length, value.Length, value, 0, value.Length) == 0);
+                    return this.Length < value.Length ?
+                        false :
+                        SpanHelpers.SequenceEqual(
+                            ref Unsafe.As<char, byte>(ref Unsafe.Add(ref this.GetRawStringData(), this.Length - value.Length)),
+                            ref Unsafe.As<char, byte>(ref value.GetRawStringData()),
+                            ((nuint)value.Length) * 2);
```

Comment:
Is this much better than:
```C#
this.AsSpan(Length - value.Length).Equals(value);
```
?  It'd be nice to avoid proliferation of this complicated unsafe code.  If not, could this be encapsulated into an overload of EqualsHelper, just like we have multiple CompareOrdinalHelper overloads?

Reply:
A little, but not vs the improved code readability. Changed.

Reply:
Bleh, still will have to check length before the subtract, 1 sec